### ISSUE TITLE
Inherited classes and Intellisense

### ISSUE THREAD

**Issue by Yuri05** [Open-Systems-Pharmacology/OSPSuite.ReportingEngine#524]
At the moment when user creates an instance of **inherited** class: initializer parameters of the **base** class are not shown via intellisense, so that it's hard to create an object without looking into the base class documentation.

E.g. when I create a **MeanModelWorkflow**, I need to look into the `Workflow$new()` documentation:
![grafik](https://user-images.githubusercontent.com/25061876/131248880-61f3c7fa-93c1-47d7-b140-9751d91abe42.png)![grafik](https://user-images.githubusercontent.com/25061876/131248947-682180bc-7611-45f4-bc3c-f4cd245cceee.png)

### If not: should we **explicitely** list all base class parameters in the definition of the initializer of an inherited class?
E.g. at the moment this is done for the **PopulationWorkflow**-class, so that all initializer parameters are shown.
From the code perspective it's duplication, but from the user perspective it's much easier to use:
![grafik](https://user-images.githubusercontent.com/25061876/131248841-21cef259-1ceb-4003-b0b4-221a4bb9542c.png)
https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/blob/ea2f843e29451bd6e65183e2d970e2329b15ee3b/R/population-workflow.R#L28-L50

It will become especially critical in the TLF when defining a plot configuration for different plot types. The base class (`PlotConfiguration`) has tons of parameters in the initializer, which are all not shown when creating a configuration of different plot types.

![grafik](https://user-images.githubusercontent.com/25061876/131249077-1d887b56-3a2b-4aa1-ba33-3bad45d84f93.png)  ![grafik](https://user-images.githubusercontent.com/25061876/131249152-f3a2efff-b9df-4334-9937-87ffe6542a23.png)

**Comment by msevestre** [comment]
@PavelBal Do you maybe also know of a way to do this ?

**Comment by PavelBal** [comment]
No, unfortunately not. The only way I am aware of is to list all the necessary arguments in the inherited class.

**Comment by abdullahhamadeh** [comment]
There may be a solution to this using the `formals()` function, but there is one hurdle to be resolved.

The idea is to build a closure that acts as the child class constructor.  This function first runs the parent initialize function and then an 'extended' `initialize` function that is specific to the child class.  The parent initialize function is defined outside the R6 class definition of the parent class.

For this example, we have a constructor for the parent class

```
parentInitializer <- function(par1,par2){
  print("Starting parent initializer")
  print(par1)
  print(par2)
  print("Ending parent initializer")
}
```

and an extended constructor for the child class:
```
extendedInitializer <- function(par3){
  print("Starting extended initializer")
  print(par3)
  print("Ending extended initializer")
}
```

and if we run the above described steps:
```
childInitializer <- makeChildInitializer(parentInitializer,extendedInitializer)
```
we arrive at a function `childInitializer`  that takes, as inputs, `par1`, `par2`, and `par3`:
![image](https://user-images.githubusercontent.com/18409098/145065194-826ca11a-aad7-49a9-901f-bc534e1b1a33.png)

To do this, the closure `childInitializer` gets knowledge of the names of arguments of `parentInitializer` and `extendedInitializer` via the `makeChildInitializer` function (i.e. the closure will retain knowledge of variables present in the environment where it is defined, namely the `makeChildInitializer` function body).   Within the `makeChildInitializer` function, these argument names are called `parentInitializer.var.names` and `extendedInitializer.var.names`.

The problem occurs when we define the child class:
```
ChildClass <- R6::R6Class(classname = "ChildClass",
                          inherit = ParentClass,
                          public = list(
                            par3 = NULL,
                            initialize = makeChildInitializer(parentInitializer,extendedInitializer))
)
```
Here, the child constructor shows the inputs to both the parent and extended initializers:
![image](https://user-images.githubusercontent.com/18409098/145067835-40253b6f-a397-4857-a64e-4903302c21a7.png)


However, the problem is that knowledge of `parentInitializer.var.names` and `extendedInitializer.var.names` is lost when the initialize function is defined in this way (ie within the R6 definition), so we end up with the error:
![image](https://user-images.githubusercontent.com/18409098/145067445-c462097e-3f3a-41a0-9c9d-8c43eb0d7ba8.png)

Code attached.   Advice welcome! 

[inheritance-intellisense.zip](https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/files/7669744/inheritance-intellisense.zip)

**Comment by abdullahhamadeh** [comment]
Can be solved using `eval(parse(...))`...

![image](https://user-images.githubusercontent.com/18409098/145098921-15825ff1-3f03-431e-8099-c178a2cc5801.png)

![image](https://user-images.githubusercontent.com/18409098/145099098-51e13b78-f537-4e41-96d1-6150a9845948.png)

Code attached.  

TBD: which classes this should be used for

[inheritance-intellisense2.zip](https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/files/7671282/inheritance-intellisense2.zip)
